### ISSUE TITLE
Simplify required PSPs and add hostPorts required to operator Chart

### ISSUE THREAD

**Issue by lewismarshall** [rook/rook#2852]
**Is this a bug report or feature request?**
* Bug Report

**Deviation from expected behavior:**
The Chart defined PSP which seems intended for Ceph clusters, specifies hostNetworking but doesn't specify the required hostPorts.

The [rook docs for Cluster CRD PSP](https://github.com/rook/rook/blob/master/Documentation/rbac.md#rbac-for-podsecuritypolicies) is almost identical to the the PSP in the chart and includes the hostPorts - although easily missed in documentation. The documentation also suggests naming the PSP `privileged` which clashes with another PSP in our clusters and was part of the motivation to re-use the PSP created by the operator Chart.

**Expected behavior:**
The hostPorts should be specified in the charts:
```
hostPorts:
    # Ceph ports
    - min: 6789
      max: 7300
    # Ceph MGR Prometheus Metrics
    - min: 9283
      max: 9283
```

Maybe there should also be a Chart value to enable hostNetwork in a block (given as you can only have a single rook-ceph cluster operating when it's enabled anyway)?

The hostPort requirement is probably also relevant to #2700 and the implementation #2820 but also #2109.

**How to reproduce it (minimal and precise):**

1. Created an operator using the [helm chart YAML](https://github.com/rook/rook/blob/master/cluster/charts/rook-ceph/) and the [specific PSP](https://github.com/rook/rook/blob/master/cluster/charts/rook-ceph/templates/psp.yaml).
1. Updated the examples to refer to the `rook-ceph-system-psp-user` for the cluster namespace (required for the Mon pods specifically):
    ```
    # Allow the default serviceAccount to use the privileged PSP
    apiVersion: rbac.authorization.k8s.io/v1
    kind: RoleBinding
    metadata:
      name: rook-default-psp
      namespace: rook-ceph
    roleRef:
      apiGroup: rbac.authorization.k8s.io
      kind: ClusterRole
      name: rook-ceph-system-psp-user
    subjects:
    - kind: ServiceAccount
      name: default
      namespace: rook-ceph
    ```
After the Mons failed to schedule, manually updated [PSP](https://github.com/rook/rook/blob/master/cluster/charts/rook-ceph/templates/psp.yaml) to include the hostPorts (specified above).

**Environment**:
* OS (e.g. from /etc/os-release): `Container Linux by CoreOS`
* Kernel (e.g. `uname -a`): `Linux dsp-ops3 4.14.96-coreos-r1 #1 SMP Tue Feb 12 21:16:22 -00 2019 x86_64 Intel(R) Xeon(R) CPU E5-2680 v4 @ 2.40GHz GenuineIntel GNU/Linux`
* Cloud provider or hardware configuration: Metal (Cisco UCS)
* Rook version (use `rook version` inside of a Rook Pod): v0.9.3
* Kubernetes version (use `kubectl version`): v.1.12.3
* Kubernetes cluster type (e.g. Tectonic, GKE, OpenShift): Custom PXE boot
* Storage backend status (e.g. for Ceph use `ceph health` in the [Rook Ceph toolbox](https://rook.io/docs/Rook/master/toolbox.html)): - Wasn't healthy (`ceph status` did not return until after the PSP updated).

**Comment by kfox1111** [comment]
I hit this too. please reopen.

ook-ceph-mon-a-canary fails to start:
message: 'pods "rook-ceph-mon-a-canary-7555fb4c68-" is forbidden: unable to validate
      against any pod security policy: [spec.securityContext.hostNetwork: Invalid
      value: true: Host network is not allowed to be used spec.volumes[2]: Invalid
      value: "hostPath": hostPath volumes are not allowed to be used spec.volumes[3]:
      Invalid value: "hostPath": hostPath volumes are not allowed to be used spec.containers[0].hostPort:
      Invalid value: 6789: Host port 6789 is not allowed to be used. Allowed ports:
      [0]]'

It does not look to be specifying a serviceAccount for the pod.

**Comment by kfox1111** [comment]
mgr's are failing to deploy too:
 Warning  FailedCreate  9m (x18 over 19m)  replicaset-controller  Error creating: pods "rook-ceph-mgr-a-7fd858476c-" is forbidden: unable to validate against any pod security policy: [spec.securityContext.hostNetwork: Invalid value: true: Host network is not allowed to be used spec.volumes[2]: Invalid value: "hostPath": hostPath volumes are not allowed to be used spec.containers[0].hostPort: Invalid value: 6800: Host port 6800 is not allowed to be used. Allowed ports: [0] spec.containers[0].hostPort: Invalid value: 9283: Host port 9283 is not allowed to be used. Allowed ports: [0] spec.containers[0].hostPort: Invalid value: 7000: Host port 7000 is not allowed to be used. Allowed ports: [0]]
[root@p0 ~]# kubectl get rs -n rook-ceph rook-ceph-mgr-a-7fd858476c
NAME                         DESIRED   CURRENT   READY   AGE
rook-ceph-mgr-a-7fd858476c   1         0         0       30m
[root@p0 ~]# kubectl get rs -n rook-ceph rook-ceph-mgr-a-7fd858476c -o yaml | grep -i serviceaccount
      serviceAccount: rook-ceph-mgr
      serviceAccountName: rook-ceph-mgr

**Comment by kfox1111** [comment]
And the osd prep jobs:
  Warning  FailedCreate  69s (x4 over 2m19s)  job-controller  Error creating: pods "rook-ceph-osd-prepare-xxxx-" is forbidden: unable to validate against any pod security policy: [spec.securityContext.hostNetwork: Invalid value: true: Host network is not allowed to be used spec.volumes[0]: Invalid value: "hostPath": hostPath volumes are not allowed to be used spec.volumes[2]: Invalid value: "hostPath": hostPath volumes are not allowed to be used spec.volumes[4]: Invalid value: "hostPath": hostPath volumes are not allowed to be used spec.volumes[5]: Invalid value: "hostPath": hostPath volumes are not allowed to be used spec.containers[0].securityContext.privileged: Invalid value: true: Privileged containers are not allowed]
looks to be:
serviceAccount: rook-ceph-osd

**Comment by travisn** [comment]
@kfox1111 You're installing through the helm chart? Are you setting [pspEnabled: true](https://github.com/rook/rook/blob/master/cluster/charts/rook-ceph/values.yaml#L49)? It doesn't seem to be picking up the [PSP settings](https://github.com/rook/rook/blob/master/cluster/charts/rook-ceph/templates/psp.yaml).

**Comment by kfox1111** [comment]
I haven't overridden it and in the valuesfile in the chart, its:
pspEnable: true

**Comment by kfox1111** [comment]
kubectl get psp | grep rook
00-rook-ceph-operator   true    SYS_ADMIN                     RunAsAny   RunAsAny           RunAsAny    RunAsAny    false            configMap,downwardAPI,emptyDir,persistentVolumeClaim,secret,projected,hostPath,flexVolume

**Comment by kfox1111** [comment]
I see the psp installed. I see a clusterrole named psp:rook 'use'ing it. I see some clusterroles binding various csi driver sa's to it. But not the mons/osds/mgr's/etc.

**Comment by travisn** [comment]
I see, the bindings to the service accounts must be missing in the helm chart, this should be fixed. Have you tried installing rook without the helm chart? I see the psp bindings in common.yaml

**Comment by kfox1111** [comment]
I have not. I prefer to use helm charts as much as possible. it simplifies the management greatly. I've worked around it for now by manually adding rolebindings for the serviceaccounts in question above.

**Comment by travisn** [comment]
got it, good to hear you could work around it.

**Comment by kfox1111** [comment]
I had to add the rolebinding to default, rook-ceph-osd, and rook-ceph-osd.

**Comment by jmolmo** [comment]
@Travisn: I cannot achieve to reproduce this issue. (with `rbacEnable` and `pspEnable` set to  true).

Apart of that, i have found that the following psp rolebindings can be found in common.yaml and not in the psp section of the clusterrolebindings: (more than the 3 reported by @kfox1111 )

rook-ceph-system-psp
rook-ceph-default-psp
rook-ceph-osd-psp
rook-ceph-mgr-psp
rook-ceph-cmd-reporter-psp

**Comment by travisn** [comment]
@jmolmo @kfox1111 Actually this issue with the helm chart is better tracked with #3851. 

The original request for this issue I believe has actually been fixed already. The PSPs are included in the helm chart now since the last release. @lewismarshall Please reopen if the intention of the issue was not fixed in v1.1. Thanks!
